phpBB/phpbb/language/language.php

```diff
@@ -0,0 +1,571 @@
+
+			case 10:
+				/**
+				 * Families: Slavic (Slovenian, Sorbian)
+				 * 1 - ends in 01: 1, 101, 201, ...
+				 * 2 - ends in 02: 2, 102, 202, ...
+				 * 3 - ends in 03-04: 3, 4, 103, 104, 203, 204, ...
+				 * 4 - everything else: 0, 5, 6, 7, 8, 9, 10, 11, ...
+				 */
+				return ($number % 100 === 1) ? 1 : (($number % 100 === 2) ? 2 : ((($number % 100 === 3) || ($number % 100 === 4)) ? 3 : 4));
+
+			case 11:
+				/**
+				 * Families: Celtic (Irish Gaeilge)
+				 * 1 - 1
+				 * 2 - 2
+				 * 3 - is 3-6: 3, 4, 5, 6
+				 * 4 - is 7-10: 7, 8, 9, 10
+				 * 5 - everything else: 0, 11, 12, ...
+				 */
+				return ($number === 1) ? 1 : (($number === 2) ? 2 : (($number >= 3 && $number <= 6) ? 3 : (($number >= 7 && $number <= 10) ? 4 : 5)));
+
+			case 12:
+				/**
+				 * Families: Semitic (Arabic)
+				 * 1 - 1
+				 * 2 - 2
+				 * 3 - ends in 03-10: 3, 4, ... 10, 103, 104, ... 110, 203, 204, ...
+				 * 4 - ends in 11-99: 11, ... 99, 111, 112, ...
+				 * 5 - everything else: 100, 101, 102, 200, 201, 202, ...
+				 * 6 - 0
+				 */
+				return ($number === 1) ? 1 : (($number === 2) ? 2 : ((($number % 100 >= 3) && ($number % 100 <= 10)) ? 3 : ((($number % 100 >= 11) && ($number % 100 <= 99)) ? 4 : (($number != 0) ? 5 : 6))));
+
+			case 13:
+				/**
+				 * Families: Semitic (Maltese)
+				 * 1 - 1
+				 * 2 - is 0 or ends in 01-10: 0, 2, 3, ... 9, 10, 101, 102, ...
+				 * 3 - ends in 11-19: 11, 12, ... 18, 19, 111, 112, ...
+				 * 4 - everything else: 20, 21, ...
+				 */
+				return ($number === 1) ? 1 : ((($number === 0) || (($number % 100 > 1) && ($number % 100 < 11))) ? 2 : ((($number % 100 > 10) && ($number % 100 < 20)) ? 3 : 4));
+
+			case 14:
+				/**
+				 * Families: Slavic (Macedonian)
+				 * 1 - ends in 1: 1, 11, 21, ...
+				 * 2 - ends in 2: 2, 12, 22, ...
+				 * 3 - everything else: 0, 3, 4, ... 10, 13, 14, ... 20, 23, ...
+				 */
+				return ($number % 10 === 1) ? 1 : (($number % 10 === 2) ? 2 : 3);
+
+			case 15:
+				/**
+				 * Families: Icelandic
+				 * 1 - ends in 1, not 11: 1, 21, 31, ... 101, 121, 131, ...
+				 * 2 - everything else: 0, 2, 3, ... 10, 11, 12, ... 20, 22, ...
+				 */
+				return (($number % 10 === 1) && ($number % 100 != 11)) ? 1 : 2;
+		}
+	}
+
+	/**
+	 * Returns language fallback data
+	 *
+	 * @return array
+	 */
+	protected function set_fallback_array()
+	{
+		$fallback_array = array();
+
+		if ($this->user_language !== false)
+		{
+			$fallback_array[] = $this->user_language;
+		}
+
+		if ($this->default_language !== false)
+		{
+			$fallback_array[] = $this->default_language;
+		}
+
+		$fallback_array[] = self::FALLBACK_LANGUAGE;
+
+		$this->language_fallback = $fallback_array;
+	}
+
+	/**
+	 * Load core language file
+	 *
+	 * @param string	$component	Name of the component to load
+	 *
+	 * @return null
+	 */
+	protected function load_core_file($component)
+	{
+		// Check if the component is already loaded
+		if (in_array($component, $this->loaded_language_sets['core']))
```

Comment:
I prefer to store the information as key value: `if (isset($this->loaded_language_sets['core'][$component])` (this way it could be also possible to store the state of the component (not found or invalid file per example))